bit_manipulation/gray_code.cpp

```diff
@@ -0,0 +1,60 @@
+#include <cassert>
+#include <iostream>
+#include <vector>
```

Comment:
please follow the outlined documentation standards defined in the contribution.md file

Reply:
still needs documentation as specified in the contribution.md file please check again. sorry for the long wait